src/pyqasm/cli/unroll.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2025 qBraid
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Script to unroll OpenQASM files
+
+"""
+
+import logging
+import os
+from pathlib import Path
+from typing import Optional
+
+import typer
+from rich.console import Console
+
+from pyqasm import dumps, load
+from pyqasm.exceptions import QasmParsingError, UnrollError, ValidationError
+
+logger = logging.getLogger(__name__)
+logger.propagate = False
+
+
+# pylint: disable-next=too-many-locals,too-many-statements
+def unroll_qasm(
+    src_paths: list[str],
+    skip_files: Optional[list[str]] = None,
+    overwrite: bool = False,
+    output_path: Optional[str] = None,
+) -> None:
+    """Unroll OpenQASM files"""
+    skip_files = skip_files or []
+
+    failed_files: list[tuple[str, Exception]] = []
+    successful_files: list[str] = []
+
+    console = Console()
+
+    def should_skip(filepath: str, content: str) -> bool:
+        if filepath in skip_files:
+            return True
+
+        skip_tag = "// pyqasm: ignore"
```

Comment:
Nice idea! Let's make it a more specific one - 
```suggestion
        skip_tag = "// pyqasm disable: unroll"
```
and can also add it directly in `QasmModule` and `validate` cli as well